### DISCUSSION TITLE
Alt + Tab sometimes cause the game no longer render to screen on fullscreen mode

### DISCUSSION THREAD

**Discussion by nahkd123** [ppy/osu#16110] [Q&A]
This one appears to be happens randomly, but basically when I press Alt + Tab to switch window while the game is running in fullscreen mode, sometimes the game just no longer render to screen anymore when I tab back in. There's no crash, unlike previous issues.

There're 2 different events in Event Viewer that might associated with it: The one with error level, which is just .NET application failed to find JSON file, and another one with warning level related to display:

```xml
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
<System>
  <Provider Name="Display" /> 
  <EventID Qualifiers="0">4101</EventID> 
  <Version>0</Version> 
  <Level>3</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2021-12-16T05:49:04.4612851Z" /> 
  <EventRecordID>74671</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>System</Channel> 
  <Computer>DESKTOP-V99B325</Computer> 
  <Security /> 
</System>
<EventData>
  <Data>igfx</Data> 
  <Data /> 
  </EventData>
</Event>
```

The displayed message in General tab only shows: ``Display driver igfx stopped responding and has successfully recovered``, so this could be related to my display driver (although other applications that uses exclusive fullscreen rarely occur this problem).

There's no interesting stuff in the game log, but if you're asking for it, here it is:
[database.log](https://github.com/ppy/osu/files/7725351/database.log)
[network.log](https://github.com/ppy/osu/files/7725352/network.log)
[performance.log](https://github.com/ppy/osu/files/7725353/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7725354/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7725356/updater.log)

Right now I can just play the game in borderless mode, or restart the game when this happens.

**Comment by peppy** [comment]
Do you have nvidia shadowplay / experience running?

Also are you graphic drivers updated to the latest version? You may need to obtain them directly from intel as windows update doesn't always have the latest version.

**Reply by nahkd123** [reply]
> Do you have nvidia shadowplay / experience running?

No. Also my laptop only have integrated GPU

> Also are you graphic drivers updated to the latest version?

My graphic driver is currently on the latest version (20.19.15.4531 to be exact). Also it looks like Intel only provides 15.x from their website.
